invisible_cities/core/fit_functions_test.py

```diff
@@ -286,6 +286,18 @@ def test_fit_with_errors(reduced):
     assert_allclose(f.values, pars)
 
 
+def test_fixed_parameters():
+    pars = np.array([3.0,  2.0, 0.50])
```

Comment:
Is it necessary to wrap this in `np.array`? Won't any sequence do?

Reply:
I see: `seeds` which you create by slicing `pars` needs to be an array. But `pars` itself does not.

Reply:
Aaah, copy 'n' paste programming ... which is exactly how code accumulates lots of pointessly misleading noise :-)

Actually, there is a serious point: *careful* thought is required when programming.

Reply:
I have to admit that this seems like a minor point, I had thought myself out doing the coding itself and as that which is written by @gonzaponte in tests tends to be the accepted correct way to do things I chose the copy 'n' paste from my betters over waste more time with boring things

Reply:
Most *single* instances of this sort of thing are indeed minor. The problem is when thoughtless copy 'n' paste programming is endemic and becomes standard practice for everything that everybody writes ... as is the case in most HEP codes. Down the years, hundreds of individually insignificant thoughtless copy 'n' pastes accumulate to make roughly 80% of the code pure, distracting, noise. Then you end up with, incomprehensible, crappy, impenetrable, buggy, pointlessly slow code.

You can't do physics today without programming. If you think that this crucial part of your professional activity is 'boring' and not worth doing well, then you're setting yourself up for pain followed by failure to achieve your goals.

I trust that you think that it is worthwhile to build detectors carefully and thoughtfully, as I trust that you think that it is worthwhile doing mathematical analyses carefully and thoughtfully. Papers you write deserve to be carefully written. The code which enables to get your work done is the same.

Reply:
> As a coding pleb I am not sensitive to these things.

Being a coding pleb is a natural starting point for all humans, and it's not an insurmountable problem as long as there's a will to, slowly, gradually, decrase one's plebishness.

Being a coding pleb and proud of it, is a serious problem for any project in which one takes part which requires notrivial amounts of programming.

Reply:
Much as it's probably not worthwhile to get into this discussion...

At no point did I say nor imply that the coding was either boring nor pointless. One small line of code which I needed to test if the actual function which I had been working on worked seemed pointless to waste time on especially since it was taken from another test which, I assumed, had been through review is the offending thing here.

I'll try not to do it again, but, in reply to your other comment:

Being strict with reviews is great but being passive aggressive and expecting perfection from the offset is not

Reply:
> At no point did I say nor imply that the coding was either boring nor pointless.

My profound apologies. You did not imply it, I erroneously inferred it from your use of the words 'waste more time with boring things'.

> Being strict with reviews is great but being passive aggressive and expecting perfection from the offset is not

I don't have any problems with your first version of the code, and I certainly don't expect perfection. But I do want to *understand* how the code works, which involves being curious about the things that appear in it. I am asking questions because you, as author, might know the answer, and thus save me precious time.

Please understand that code review is not an attack on your person, but a collaborative effort to improve not only the code but also both your and my skill at implementing stuff.

When I make comments about how things can be done better, it is not because I expected you to do it that way in the first place, but because I think you (and people who use and read your code) might benefit from it in the future. So please try to take it as an act of collaboration rather than aggression. I like to think that in technical circles we can cut the crap and make technical comments without pussyfooting around in case someone interprets technical comments as personal attacks.

Reply:
I don't interpret anything as a personal attack. I would, however, prefer that a comment about one small section in a test not be interpreted as my disdain for the project. I actually quite enjoy coding even if I'm more interested in the solving the problem than the specifics. Maybe I'm too sensitive, too many years of everything I say being passed through amor/odio filters.

Please feel free to give me homework, I learn better by doing, I can make changes to try and make the code clearer etc.

Reply:
> Please feel free to give me homework,

I have to do my homework on this, before I can give you any sensible homework.